concrete-csprng/src/counter/mod.rs

```diff
@@ -256,64 +279,71 @@ impl<G: AesBatchedGenerator> AesCtrGenerator<G> {
         &self.state
     }
 
-    /// Returns the bound of the generator if any.
-    pub fn get_bound(&self) -> Option<&State> {
-        self.bound.as_ref()
-    }
-
-    /// Returns whether the generator is bounded.
-    pub fn is_bounded(&self) -> bool {
-        self.bound.is_some()
+    /// Returns the bound of the generator.
+    pub fn get_bound(&self) -> &State {
+        &self.bound
     }
 
-    /// Returns the number of remaining bytes if bounded
-    pub fn remaining_bytes(&self) -> Option<usize> {
-        self.bound.as_ref().map(|bound| {
-            let res = ((bound.aes_ctr.0 - self.state.aes_ctr.0) as i128) * 16
-                + (bound.byte_ctr.0 as i128 - self.state.byte_ctr.0 as i128);
-            res as usize
-        })
+    /// Returns the number of remaining bytes
+    pub fn remaining_bytes(&self) -> u128 {
+        let step = self.step.0 as u128;
+
+        // Computes the difference in bytes between the two AES Counter
+        let res_1 = ((self.bound.aes_ctr.0 - self.state.aes_ctr.0) / step).saturating_mul(16);
+        if self.bound.byte_ctr.0 > self.state.byte_ctr.0 {
+            // If the difference between the byte counter of the bound and the one of self
+            // is positive, adds it to res_1.
+            // This will saturate nicely.
+            res_1.saturating_add((self.bound.byte_ctr.0 - self.state.byte_ctr.0) as u128 / step)
+        } else {
+            // Otherwise, substracts the opposite of the difference.
+            let mut res_2 = (self.state.byte_ctr.0 - self.bound.byte_ctr.0) as u128 / step;
+            if res_1 == u128::MAX {
```

Comment:
This should have a small example/explanation on the potential problem arising when res_1 has already saturated.